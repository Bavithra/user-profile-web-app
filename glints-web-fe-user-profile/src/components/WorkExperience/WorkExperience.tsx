import React from "react";
import { InputFields } from "../../enums/InputFields.enum";
import { InputContainer } from "../../pages/ProfilePage/ProfilePage.styles";
import { WorkExperience } from "../../types/WorkExperience.type";
import { WorkDetail } from "./WorkExperience.styles";

type Props = {
  workExperience: WorkExperience;
};

export default function WorkExperienceItem(props: Props) {
  const { workExperience } = props;

  return (
    <InputContainer>
      <WorkDetail>
        <div>{InputFields.StartDate}</div>
        <div>{workExperience["start-date"]}</div>
      </WorkDetail>
      <WorkDetail>
        <div>{InputFields.EndDate}</div>
        <div>{workExperience["end-date"]}</div>
      </WorkDetail>
      <WorkDetail>
        <div>{InputFields.Company}</div>
        <div>{workExperience.company}</div>
      </WorkDetail>
      <WorkDetail>
        <div>{InputFields.JobTitle}</div>
        <div>{workExperience["job-date"]}</div>
      </WorkDetail>
    </InputContainer>
  );
}
