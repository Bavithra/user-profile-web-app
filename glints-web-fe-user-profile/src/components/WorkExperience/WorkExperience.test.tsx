import React from "react";
import renderer from "react-test-renderer";
import { WorkExperience } from "../../types/WorkExperience.type";
import WorkExperienceItem from "./WorkExperience";

describe("WorkExperienceItem Block", () => {
  it("renders correctly", () => {
    let workExperience: WorkExperience = {
      "start-date": "",
      "end-date": "",
      "job-date": "",
      company: "",
      "company-logo": "",
      "job-description": "",
    };
    const tree = renderer
      .create(<WorkExperienceItem workExperience={workExperience} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
