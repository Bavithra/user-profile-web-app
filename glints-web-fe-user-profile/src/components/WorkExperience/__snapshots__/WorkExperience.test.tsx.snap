// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkExperienceItem Block renders correctly 1`] = `
<div
  className="sc-dkzDqf fhLHNL"
>
  <div
    className="sc-kDDrLX eMMeIp"
  >
    <div>
      Start date
    </div>
    <div />
  </div>
  <div
    className="sc-kDDrLX eMMeIp"
  >
    <div>
      End date
    </div>
    <div />
  </div>
  <div
    className="sc-kDDrLX eMMeIp"
  >
    <div>
      Company
    </div>
    <div />
  </div>
  <div
    className="sc-kDDrLX eMMeIp"
  >
    <div>
      Job title
    </div>
    <div />
  </div>
</div>
`;
